.github/actions/setup-python-poetry/action.yaml

```diff
@@ -0,0 +1,47 @@
+name: "Install Python and Poetry"
+description: "Add Poetry, dependency manager for Python"
+
+inputs:
+  python-version: # Dynamically set Python version
+    description: "Python version"
+    required: true
+  poetry-version: # Dynamically set Poetry version
+    description: "Poetry version"
+    required: true
+  install-deps: # Option to install Python dependencies
+    description: "Should the action install deps?"
+    required: false
+    default: false
+  only-dev-deps: # Option to install only dev dependencies
+    description: "Should the action only install the dev dependencies?"
+    required: false
+    default: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install Poetry ${{ inputs.poetry-version }}
+      run: |
+        curl -sSL ${{ env.POETRY_URL }} | \
+          python - --version ${{ inputs.poetry-version }}
+        echo "$HOME/.local/bin" >> $GITHUB_PATH
+      shell: bash
+      env:
+        POETRY_URL: https://install.python-poetry.org
+
+    - name: Set up Python ${{ inputs.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ inputs.python-version }}
+        cache: "poetry"
```

Comment:
What does this do? Perhaps should add a comment here to indicate what's being cached, and why?

Reply:
The `cache: "poetry"` option enables caching for Python dependencies that are managed by Poetry.

Behind the scenes, the action will create a hash based on the poetry.lock file, which contains the exact version of each dependency. This ensures that the cache is only used when the dependencies have not changed. When the workflow runs again, and the cache key matches the current poetry.lock file, the cached dependencies will be used instead of downloading and installing them from scratch.

I believe the cache is stored somewhere in Github and lives in the repository (not sure of the details about cache storage)